static/js/stream_edit.js

```diff
@@ -789,12 +699,61 @@ export function initialize() {
 
     $("#subscriptions_table").on("click", "#change-stream-privacy-button", change_stream_privacy);
 
-    $("#subscriptions_table").on("click", ".close-modal-btn", (e) => {
-        // This fixes a weird bug in which, subscription_settings hides
-        // unexpectedly by clicking the cancel button in a modal on top of it.
+    $("#subscriptions_table").on("click", "#open_stream_info_modal", (e) => {
+        e.preventDefault();
         e.stopPropagation();
+        const stream_id = get_stream_id(e.target);
+        const stream = sub_store.get(stream_id);
+        const template_data = {
+            stream_id,
+            stream_name: stream.name,
+            stream_description: stream.description,
+        };
+        const change_stream_info_modal = render_change_stream_info_modal(template_data);
+        $("#change_stream_info_modal").remove();
+        $("#subscriptions_table").append(change_stream_info_modal);
+        overlays.open_modal("#change_stream_info_modal");
+    });
+
+    $("#subscriptions_table").on("keypress", "#change_stream_description", (e) => {
+        // Stream descriptions can not be multiline, so disable enter key
+        // to prevent new line
+        if (e.key === "Enter") {
+            return false;
+        }
+        return true;
+    });
+
+    $("#subscriptions_table").on("click", "#save_stream_info", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+        const sub = get_sub_for_target(e.currentTarget);
+
+        const url = `/json/streams/${sub.stream_id}`;
+        const data = {};
+        const new_name = $("#change_stream_name").val().trim();
+        if (new_name !== sub.name) {
+            data.new_name = new_name;
+        }
+        const new_description = $("#change_stream_description").val().trim();
+        if (new_description !== sub.description) {
+            data.description = new_description;
+        }
+        const status_element = $(".stream_change_property_info");
+        overlays.close_modal("#change_stream_info_modal");
+        settings_ui.do_settings_change(channel.patch, url, data, status_element);
     });
 
+    $("#subscriptions_table").on(
+        "click",
+        ".close-modal-btn, .close-change-stream-info-modal",
+        (e) => {
+            // This fixes a weird bug in which, subscription_settings hides
+            // unexpectedly by clicking the cancel button in a modal on top of it.
+            e.stopPropagation();
```

Comment:
I feel like we shouldn't need something like this -- it feels like the bug is that the modal backdrop isn't working properly.

Perhaps @sahil839 or @ganpa3 can find the proper fix for this.

Reply:
There is one more instance of this same problem in this file.
[https://github.com/zulip/zulip/blob/054be8dd2e18a381d601200fc1d5198889ccd7a6/static/js/stream_edit.js#L790](https://github.com/zulip/zulip/blob/054be8dd2e18a381d601200fc1d5198889ccd7a6/static/js/stream_edit.js#L790)

Reply:
The modal closes due to a click handler registered in [here](https://github.com/zulip/zulip/blob/777eeb6b8f551c0c0fdeffb2334a595b40fb73f0/static/js/subs.js#L999). Either changing that (not ideal) or appending the `change_stream_info_modal` to `body` will fix this naturally.

Reply:
Can't we just tighten that selector to be `foo[data-dismiss]` for the appropriate value of `foo` that is the `data-dismiss` element for the "Stream settings" overlay?

Reply:
Yeah, Tim's suggestion about this seems to work perfectly fine for this.
I tried to find a bit more about why that line was added in `subs.js` and found that it was specifically added for the `cancel` button in stream creation form in [https://github.com/zulip/zulip/commit/1886f0a015c5c3a9141a8cbd439d0c737dde2af8](https://github.com/zulip/zulip/commit/1886f0a015c5c3a9141a8cbd439d0c737dde2af8). I think it assumed that we will have a data-dismiss for that button only inside `.subscription` but as we now have new modals (and corresponding `data-dismiss` elements for them) due to stream-edit section. So binding that selector correctly appeared like a natural fix for it.
I added a commit (https://github.com/zulip/zulip/pull/18872/commits/6084bf980f5891bc474d76476a261b1e0810e43a) just next to this so that both commits are coherent.

Reply:
> Can't we just tighten that selector to be `foo[data-dismiss]` for the appropriate value of `foo` that is the `data-dismiss` element for the "Stream settings" overlay?

Yeah, that's what I suggested.